### DISCUSSION TITLE
Impossible to build libsrc/ folder (.global seems to be broken with WSL)

### DISCUSSION THREAD

**Discussion by jedeoric** [cc65/cc65#2031] [Q&A]
Hello,

Since commit d6c37a757d784ac6691e4920c03c406ac82411c6, ".global" seems to be broken under WSL (ubuntu 22.02 or ubuntu 20.04)

# 3513342445bcb5641b80a3a3aee7605220137ed8 => before d6c37a757d784ac6691e4920c03c406ac82411c6 commit
$ git checkout 3513342445bcb5641b80a3a3aee7605220137ed8
$ make
... everything is OK

# d6c37a757d784ac6691e4920c03c406ac82411c6 introduce the problem with WSL
$ git checkout d6c37a757d784ac6691e4920c03c406ac82411c6
$ make
...
common/_directerrno.s:20: Error: Symbol '___oserror' is undefined
common/_directerrno.s:19: Error: Symbol '___seterrno' is undefined
make[2]: *** [Makefile:280 : ../libwrk/apple2/_directerrno.o] Erreur 1

# Switch to the last commit of cc65 repo :

$ git checkout 7f1dd09bcb3a0f97be7bce7c4344cdf6cb8c280a
$ make
...
common/_directerrno.s:20: Error: Symbol '___oserror' is undefined
common/_directerrno.s:19: Error: Symbol '___seterrno' is undefined

if i introduce ".export ___oserror and .export ___seterrno" in the right file with ___oserror and  ___seterrno, and i add ".import  ___oserror, ___seterrno", it corrects the thing but others .global does not works (as __heap* -> i can add .import/.export to pass the build of the libs, but it's not the way it should be done)

$uname -a
Linux 5.15.90.1-microsoft-standard-WSL2 #1 SMP Fri Jan 27 02:56:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

$gcc --version
gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0

The same problem appears with WSL/ubuntu 20.04

**Comment by mrdudz** [comment]
you'll have to do a clean build, ie "make clean" first

**Comment by jedeoric** [comment]
I did it also to test if it was that, but the same problem can be reproduce : 

# Get the main branch
$ git clone https://github.com/cc65/cc65.git
$ cd cc65 && make clean && make
apple2 - common/_directerrno.s
common/_directerrno.s:20: Error: Symbol '___oserror' is undefined
common/_directerrno.s:19: Error: Symbol '___seterrno' is undefined

# Test twice
$ make clean && make
...
common/_directerrno.s:20: Error: Symbol '___oserror' is undefined
common/_directerrno.s:19: Error: Symbol '___seterrno' is undefined

The problem appears only under WSL, because i was able to build it on another computer

**Comment by mrdudz** [comment]
that is extremely odd. i'll have to try to reproduce that (i usually work in msys on windows, not WSL)

**Comment by polluks2** [comment]
Indeed, my Debian/WSL works fine. The subsystem doesn't support `lsusb`, but cc65 requires nothing fancy.

**Comment by jedeoric** [comment]
I will test something tonight, because i was able to build cc65 with wsl under another computer

**Comment by jedeoric** [comment]
Hello, 

I found the problem.

For various reasons, i set into my .bashrc : 

CA65_INC=/usr/share/cc65/asminc/
export CA65_INC

I did not search why the https://github.com/cc65/cc65/commit/d6c37a757d784ac6691e4920c03c406ac82411c6 commit  introduces this, it seems that now it reads .inc from $CA65_INC instead of .inc from asminc compilation folder 

This does the fix : 

$ unset CA65_INC
$ make 

Since i have an folder from June 2022, the errno.inc contains in $CA65_INC folder was (__oserror instead of ___oserror): 

        .global         __errno, __oserror
        .global         __osmaperrno
        .global         __seterrno
        .global         __directerrno, __mappederrno

**Comment by mrdudz** [comment]
using those environment variables was debrecated a long time ago - you shouldnt need them :)

closing...
